ftd-host/sql.ftd

```diff
@@ -0,0 +1,305 @@
+for: $p in $people
+
+
+-- ds.markdown:
+
+Which will look something like this:
+
+
+-- show-person: $p
+for: $p in $people
+
+
+
+
+
+-- ds.h1: Passing Named Parameters in Your Query
+
+The `sql` processor allows you to pass named parameters in your queries. 
+You can pass a named parameter by defining it as a header/property of the section where the query is executed. 
+To access the named parameter in your query, use the following syntax:
+
+-- ds.code: Syntax for Named Parameters
+lang: sql
+
+\$<PARAM_NAME>::<PARAM_TYPE>
+
+-- ds.markdown:
+
+In this syntax, the name following the `$` symbol represents the parameter's name, 
+and you can specify its type by appending `::<PARAM_TYPE>` to indicate the desired data type.
+
+You can find the list of the data types that are currently supported by fastn:
+- [Supported PostgreSQL Data Types](/sql/#supported-postgresql-data-types)
+- [Supported SQLite Data Types](/sql/#supported-sqlite-data-types)
+
+-- ds.markdown:
+
+Let's illustrate this with an example. Suppose you want to fetch a user from the `users` table by their `id`:
+
+-- ds.code: Retrieving User Data by User ID
+lang: ftd
+
+\-- import: fastn/processors as pr
+
+\-- integer id: 1
+
+\-- person jack:
+$processor$: pr.sql
+id: $id
+
+SELECT * FROM users WHERE id = $id::INTEGER;
+
+\-- show-person: $jack
+
+-- ds.markdown:
+
+The result will look something like this:
+
+-- show-person: $people.0
+
+-- ds.h2: Taking input from other processors
+
+-- ds.markdown:
+
+This approach also enables you to pass a value obtained from any other processor as an input for your SQL queries. For instance, you can utilize the [`request-data`](/request-data) processor and [Dynamic URLs](/dynamic-urls/) to dynamically create user profile pages, similar to Twitter and other social networks.
+
+-- ds.code: Retrieving User Data by Username
+lang: ftd
+
+\-- import: fastn/processors as pr
+
+\-- string username:
+$processor$: pr.request-data
+
+\-- user user-data:
+$processor$: pr.sql
+username: $username
+
+SELECT * FROM users WHERE username = $username::STRING;
+
+\-- user-profile: $user-data
+
+-- ds.markdown:
+
+Now, whenever a visitor accesses your dynamic page, such as `/user/<username>`, 
+fastn will retrieve the username from the URL using the `request-data` processor and pass it to your SQL query as a named parameter. 
+This allows you to retrieve the data of the user whose username matches the passed value.
+
+
+
+
+-- ds.h1: Manipulating Data and Handling Errors
+
+You can manipulate data by executing `INSERT`, `UPDATE`, and `DELETE` queries. Unlike the `SELECT` query, 
+these queries do not return a result. Instead, they return the query's `status` after execution. 
+This `status` informs you whether the query was successful or if it encountered an error. 
+The `status` can have either an `integer` or `string` data type.
+
+If the `status` is set as an `integer` data type, the query will return an exit code of 0 (indicating success) or 1, 
```

Comment:
This is wrong. INSERT etc can return stuff using RETURNING. Delete all this, this is database specify stuff, there is no SQL specific answer here.

Tell we do not have any error handling right now. What if database is down, or if query uses wrong table name etc, error handling is to handle these stuff, and it is impossible to handle them right now, so answer is we have no error handling.

Reply:
Removing the Error handling section for now, but we need to consider an alternative approach because the `RETURNING` keyword is only supported by a few databases such as PostgreSQL, and it is not supported in SQLite or MySQL (which we currently do not support). The `status` flag was added to provide users with basic information about the status of their query, whether it was a `success` or `failure`. Users can still use the `RETURNING` keyword with `pg` to obtain the result if they wish.